### PR TITLE
postgres: use jackc/pgx/v5 instead of lib/pq

### PR DESCRIPTION
**Pull request by achille-roussel**
@johanbrandhorst suggested we switch to `jackc/pgx` because `lib/pq` may not be maintained anymore.

### CODE REVIEW

File: wasip1/net_wasip1_test.go [hunk 1]
```diff
@@ -35,10 +32,6 @@ func TestConn(t *testing.T) {
 			network: "tcp6",
 			address: ":0",
 		},
-		{
-			network: "unix",
-			address: unixSocket,
-		},
```
Comment (achille-roussel) [inline wasip1/net_wasip1_test.go:41]: This isn't working anymore on `gotip` so I removed it for now.

**Review by chriso — APPROVED** [review]

File: wasip1/net_wasip1_test.go (hunk 1 above)
Comment (chriso) [reply wasip1/net_wasip1_test.go:41]: Any idea why? Is it a regression we need to submit a fix for?

File: postgres/go.mod [hunk 2]
```diff
@@ -3,8 +3,15 @@ module github.com/stealthrocket/net/postgres
...
 )
 
+require (
+	github.com/jackc/pgpassfile v1.0.0 // indirect
+	github.com/jackc/pgservicefile v0.0.0-20221227161230-091c0ba34f0a // indirect
+	golang.org/x/crypto v0.9.0 // indirect
+	golang.org/x/text v0.9.0 // indirect
+)
+
 replace github.com/lib/pq => github.com/stealthrocket/pq v0.0.0-20230609191514-73472796856a
```
Comment (johanbrandhorst) [inline postgres/go.mod:17]: I think we can remove this replace now too?

File: postgres/postgres_wasip1_test.go [hunk 3]
```diff
@@ -3,28 +3,39 @@
...
 
-	"github.com/lib/pq"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/stdlib"
 	"github.com/stealthrocket/net/wasip1"
 )
 
+func init() {
+
+}
```
Comment (johanbrandhorst) [inline postgres/postgres_wasip1_test.go:18]: Is this needed?

File: postgres/go.mod (hunk 2 above)
Comment (achille-roussel) [reply postgres/go.mod]: Thanks! I missed it :)

File: postgres/postgres_wasip1_test.go (hunk 3 above)
Comment (achille-roussel) [reply postgres/postgres_wasip1_test.go]: Nope

**Review by johanbrandhorst — APPROVED** [review]

File: wasip1/net_wasip1_test.go (hunk 1 above)
Comment (achille-roussel) [reply wasip1/net_wasip1_test.go:41]: I haven't had time to look closely into it yet.

Comment (achille-roussel) [reply wasip1/net_wasip1_test.go:41]: Those are the errors:

```
--- FAIL: TestConn (1.17s)
    --- FAIL: TestConn/unix (0.00s)
        --- FAIL: TestConn/unix/BasicIO (0.00s)
            conntest.go:32: unable to make pipe: dial :0 :0: dial :0 :0: getaddrinfo: Invalid argument
        --- FAIL: TestConn/unix/PingPong (0.00s)
            conntest.go:33: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/RacyRead (0.00s)
            conntest.go:34: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/RacyWrite (0.00s)
            conntest.go:35: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/ReadTimeout (0.00s)
            conntest.go:36: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/WriteTimeout (0.00s)
            conntest.go:37: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/PastTimeout (0.00s)
            conntest.go:38: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/PresentTimeout (0.00s)
            conntest.go:39: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/FutureTimeout (0.00s)
            conntest.go:40: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/CloseTimeout (0.00s)
            conntest.go:41: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
        --- FAIL: TestConn/unix/ConcurrentMethods (0.00s)
            conntest.go:42: unable to make pipe: listen unix /tmp/TestConn3414946857/001/wasip1.sock: bind: Address already in use
```

Comment (chriso) [reply wasip1/net_wasip1_test.go:41]: Unix sockets on the file system aren't removed when you close them. That might be it? We just need to remove them explicitly after each test is done.

**Pull request merged**